doc/api/esm.md

```diff
@@ -239,11 +241,35 @@ const { default: barData } =
   await import('./bar.json', { assert: { type: 'json' } });
 ```
 
-Node.js supports the following `type` values:
+Node.js supports the following `type` values, for which the assertion is
+mandatory:
+
+| Assertion `type` | Needed for       |
+| ---------------- | ---------------- |
+| `'json'`         | [JSON modules][] |
+| `'webassembly'`  | [Wasm modules][] |
+
+Imports of CommonJS and ES module JavaScript files, and Node.js core modules, do
+not need an import assertion; but they can be given a `type` of `'javascript'`
+if desired:
+
+```js
+const url = getUrlOfModuleToBeDynamicallyImported();
+const { protocol, pathname } = new URL(url, import.meta.url);
+
+let assertType = 'javascript';
+if (protocol === 'data:' ?
+  pathname.startsWith('application/json') :
+  pathname.endsWith('.json')) {
+  assertType = 'json';
+} else if (protocol === 'data:' ?
+  pathname.startsWith('application/wasm') :
+  pathname.endsWith('.wasm')) {
+  assertType = 'webassembly';
+}
```

Comment:
Those ternaries inside a the `if` conditions are nuts. It's probably the most concise, but please don't do that and expand them instead.

If nothing else, is very difficult even just to grok

```suggestion
if (
	protocol === 'data:' ?
  pathname.startsWith('application/json') :
  pathname.endsWith('.json')
) {
  assertType = 'json';
} else if (
	protocol === 'data:' ?
  pathname.startsWith('application/wasm') :
  pathname.endsWith('.wasm')
) {
  assertType = 'webassembly';
}
```

Reply:
I agree; the ternary should be moved down to the assignment instead of in the `if`

Reply:
The ternaries in the conditions were @aduh95’s suggestion 😄 

It looks like all you’re suggesting I change is essentially adding whitespace? The new version still keeps ternaries in conditional blocks.

Reply:
Personally I don't like the new version either. i'd probably suggest intermediate variables rather than making a complex branching construct.

Reply:
@GeoffreyBooth my code suggestion was just a "fallback" if you're married to the ternaries within the `if`. But I would much rather they be moved out as Jordan described.

@ljharb I don't like the suggestion, but if it's the best I can get, better than nothing 😜